### PR TITLE
Fixed #26140 -- Suppressed MySQL warning when inserting binary content

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

**Comment by claudep** [comment]
I'm not completely sure about the implementation. I could have tweaked the MySQL compiler. Another question is about making `binary_placeholder` a method instead of a class variable. Will we need some parameter in the future?
I could also have written something more generic instead of being binary-specific.

**Comment by claudep** [comment]
One of that question is answered by a test failure (_binary should not be specified for None).

**Comment by timgraham** [comment]
@mjtamlyn @akaariai @jarshwah, any thoughts about this?

File: tests/model_fields/tests.py [hunk 1]
```diff
@@ -860,17 +862,20 @@ class BinaryFieldTests(test.TestCase):
...
-            dm.save()
-            dm = DataModel.objects.get(pk=dm.pk)
-            self.assertEqual(bytes(dm.data), bytes(bdata))
-            # Resave (=update)
-            dm.save()
-            dm = DataModel.objects.get(pk=dm.pk)
-            self.assertEqual(bytes(dm.data), bytes(bdata))
-            # Test default value
-            self.assertEqual(bytes(dm.short_data), b'\x08')
+        with captured_stderr() as stderr:
```
Comment (timgraham) [inline tests/model_fields/tests.py]: I think I'd skip the test modifications as it seems like it may be too fragile.

**Comment by iterq1** [comment]
@claudep
Hi! Does this fix work for SELECT queries? I noticed that `get_placeholder` used only in `SQLUpdateCompiler`, and `SQLInsertCompiler`, but for SELECT queries `SQLCompiler` is used.

**Comment by timgraham** [comment]
According to the ticket, the warning happens on INSERT queries. Do you see a warning on SELECT queries? If so, please [open a ticket](https://code.djangoproject.com/) with details.

**Comment by iterq1** [comment]
@timgraham I'm afraid that I can't open a ticket, 'cause I'm using Django 1.6. 1.8 is the earliest version allowed. So I just wanted to be sure that this fix works for SELECT queries too, maybe I missed somethig in the source code.
But as this fix was added only in 1.10, should I open a ticket with not propriate version?

**Comment by timgraham** [comment]
Django 1.6 is unsupported. This fix won't be incorporated in any other versions of Django.

**Pull request merged**